### DISCUSSION TITLE
TiVo remote configuration - is remote address configurable?

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#191] [Q&A] [answered]
I'm trying to anticipate issues with getting everything set up next week, when I will be in the same physical location as my Slingboxes and my router. One of the issues I remember having when first setting my Slingboxes up via Slingplayer Desktop/Web was the remote address that gets assigned to the TiVo remote. I have 4 different TiVos all co-located in the same room, and each one has a different remote address assigned to the corresponding remote (remote addresses 2, 3, 4, and 5, if I remember correctly).

The problem with a "default" remote (and presumably, a default Slinger remote) is that the remote address defaults to "1". I took a quick look at the TiVo remote definition in the CustomRemotes directory, and I didn't see any way to configure the remote address. Did I miss something in the TiVo remote config file?

**Comment by RandomMan719** [comment]
What Tivo Model, do you have?

Here's the link to the remote database, http://autoupdate.sling.com/remotedb/Remote-keymaps/

**Reply by DVoorhees1957** [reply]
I have 4 different models: a TiVo Series 2 single tuner (model 540040) with remote address 2, a Tivo Premier XL (model 748000) with remote address 4, a Tivo Premier (model 746500) with remote address 5, and a Tivo BOLT (model 849500) with remote address 3. I assume the remote address "changes" the key code that gets sent to the corresponding TiVo unit... yes? Or maybe there's some sort of "handshake" at the start of a remote command, that pairs the remote to a specific TiVo (or TiVos) that have a matching remote address. Otherwise, how could the 4 different remotes I have be isolated to a specific TiVo device? I'm trying to avoid a situation where the Slinger code sends the same command to all 4 of my TiVos at the same time...

**Reply by RandomMan719** [reply]
It is possible, this site has different remote .bin files for tivo unit addresses, for the Tivo Series3.  It has 6 different .bin files for the same device. 

http://www.hifi-remote.com/forums/dload.php?action=file&file_id=4804

So I assume they send different codes, depending on the remote address.  Maybe like remote address 1 uses codes 0-50 and remote address 2 uses codes 51-100 and so on, just guessing.  I don't know how to look at the .bin files to see the keycodes being sent to the device.

**Comment by GerryDazoo** [comment]
My code just sends the appropriate Keycodes to the singbox in question. If it works with the Sling software it should work with mine. I have no idea what a Tivo remote address is.

**Comment by tom76017** [comment, marked as answer]
Each TiVo will have a different IP address.  Take a look at https://github.com/GerryDazoo/Slinger/discussions/30 and https://github.com/GerryDazoo/Slinger/discussions/84

I'm not sure if the older TiVo's support Network Remote but my Edge, Mini A92 and A95's do.  I actually prefer using my iPhone, iPad, or 
windows to control Slinger streams.

**Reply by DVoorhees1957** [reply]
Ah, that sounds like a cool approach... I'll look into that.

**Comment by DVoorhees1957** [comment]
After thinking about it overnight, I realized I'm asking the wrong question. Currently, the Slingbox process that ensures the right remote address gets used, and the right codes get sent from the Slingbox to the TiVo, is running Slingplayer Desktop and specifying what device is connected to the Slingbox inputs (and how that device is connected). It's during this process that you pick from the list of Tivo devices - there are multiple such devices, each with different TiVo remote addresses.

So I believe the question I should have asked is: will Slingplayer Desktop continue to run after Nov 9th, and will it continue to allow you to change the connected device and/or change how that device is connected to the Slingbox. If so, there's nothing to worry about. If not, then it makes me concerned that you won't be able to change the connected device or the connection type after the Nov 9th date...

**Reply by tom76017** [reply]
Try unplugging your modem so that you only have a local LAN with no Internet and see if it still works.

**Reply by DVoorhees1957** [reply]
Thanks, I'm away/remote, as I am most of the time, so can't run any tests for about 4-5 days... I was hoping someone had already tried some of this, since configuration via Slingplayer Web/Slingplayer Desktop was a big part of the "old way" of doing things...

**Reply by RandomMan719** [reply]
I don't think Slingplayer Web/Desktop, SlingPlayer 2.0, are going to work, they require account login, as far as I know.
Slinger allows you to connect to any input and stream from it and change external ir remotes, (change any device connected to it), with the caveat of the internal tuner, if it's not setup prior, then you won't be able to scan or tune to new frequencies.
